components/feature/autofill/src/main/java/mozilla/components/feature/autofill/AbstractAutofillService.kt

```diff
@@ -60,10 +64,33 @@ abstract class AbstractAutofillService : AutofillService() {
     }
 
     override fun onSaveRequest(request: SaveRequest, callback: SaveCallback) {
-        // This callback should not get invoked since we do not indicate that we are interested in
-        // saving any data (yet). If for whatever reason it does get invoked then we pretent that
-        // we handled the request successfully. Calling onFailure() requires to pass in a message
-        // and on Android systems before Q this message may be shown in a toast.
+        var username: String? = null
+        var password: String? = null
+        var parsedStructure: ParsedStructure? = null
+        for (fillContext in request.fillContexts) {
+            parsedStructure = parseStructure(this, fillContext.structure.toRawStructure())
+            parsedStructure?.also { ps ->
+                ps.username?.let { username = it }
+                ps.password?.let { password = it }
+            }
+        }
+        parsedStructure ?: return
+        username ?: return
+        password ?: return
+
+        @OptIn(DelicateCoroutinesApi::class)
+        GlobalScope.launch(Dispatchers.IO) {
+            val lookupDomain = parsedStructure.getLookupDomain(configuration.publicSuffixList)
+            val loginEntry = Login(
```

Comment:
You should create a `LoginEntry` instead of `Login`, then there's no need to call `toEntry()`.

The naming isn't super obvious, but one type is meant to represent a record we got back from the local database - `Login`, it has a `guid` which is generated during storage. Note how that's a none-nullable type.

`LoginEntry` is identical but without a database `guid` - representing "input data", e.g. something that's coming from the user (or a website, or an app in this case), i.e. we may or may not have a similar record in the database already.

Reply:
Thanks for clarifying!
Fixed.